typings/index.d.ts

```diff
@@ -2661,21 +2661,12 @@ declare module 'discord.js' {
     public delete(channel: StageChannel | Snowflake): Promise<void>;
   }
 
-  export class ThreadManager extends BaseManager<Snowflake, ThreadChannel, ThreadChannelResolvable> {
+  export class ThreadManager<AllowedThreadType> extends BaseManager<Snowflake, ThreadChannel, ThreadChannelResolvable> {
     constructor(channel: TextChannel | NewsChannel, iterable?: Iterable<any>);
     public channel: TextChannel | NewsChannel;
-    public create(options: {
-      name: string;
-      autoArchiveDuration: ThreadAutoArchiveDuration;
-      startMessage?: MessageResolvable;
-      type?: ThreadChannelType | number;
-      reason?: string;
-    }): Promise<ThreadChannel>;
-    public fetch(options: ThreadChannelResolvable, cacheOptions?: BaseFetchOptions): Promise<ThreadChannel | null>;
-    public fetch(
-      options?: { archived?: FetchArchivedThreadOptions; active?: boolean },
-      cacheOptions?: { cache?: boolean },
-    ): Promise<FetchedThreads>;
+    public create(options: ThreadCreateOptions<AllowedThreadType>): Promise<ThreadChannel>;
+    public fetch(options?: ThreadChannelResolvable, cacheOptions?: BaseFetchOptions): Promise<ThreadChannel | null>;
+    public fetch(options?: FetchThreadsOptions, cacheOptions?: BaseFetchOptions): Promise<FetchedThreads>;
```

Comment:
I don't understand why this was changed, force is never a valid option for this overload, it should be suggested. For the previous overload it is valid.

Reply:
It was merging those two and hence wasn't suggesting the `force` field even when `options` is a `ThreadChannelResolvable`. It isn't `invalid` tho, it is more of a **"redundant"** param when options is `FetchThreadsOptions` as it gets ignored.

Reply:
huh, cause `ThreadChannelResolvable` somehow equates to `FetchThreadsOptions` ??? ts you okay? whatever, if it works it works.

Reply:
With the above change, this could be changed back to `{ cache?: boolean }`.
At least in my testing it seems to work properly with suggesting `cache` and rejecting `{ force: true }`.

Reply:
It rejecting `force` for `FetchThreadsOptions` wasn't the issue, the problem is that it doesn't suggest `force` for `ThreadChannelResolvable` too which makes it **"seems"** like `force` isn't a valid param. It does allow it to compile so it's not that big of an issue, just a minor inconvenience. Making both of them `BaseFetchOptions` solves the issue and since `force` does nothing for the second overload, it's not a problem even if the user passes it.

Reply:
I'm unable to reproduce that behavior, I'm seeing the suggestions just fine.

Reply:
Weird, doesn't seem to work for me. Using `{ cache?: boolean }` for second overload makes it **not** suggest `force` for the first one too:
![Code_bNJDLuQZwJ](https://user-images.githubusercontent.com/51517103/124361903-5dd99600-dc4f-11eb-9381-7d9f4ad4529b.png)

Reply:
I just realized that we can solve this issue by using Conditional Types. We can merge the overloads and let conditional typings do its work:
![Code_TyyPLLLIsa](https://user-images.githubusercontent.com/51517103/124362673-10135c80-dc54-11eb-93a1-8d32adf14626.png)
I tested it and it works nicely. It suggests both `cache` & `force` when `options` is `ThreadChannelResolvable` and only `cache` when it is `FetchThreadsOptions`.

Reply:
I don't know why it's not showing up for you 🤷 
![image](https://user-images.githubusercontent.com/24881032/124362788-6e7e1200-dc37-11eb-99de-b5df69b5ef07.png)

Reply:
I don't think the `.type` check actually guards, maybe thats why.

Reply:
> 
> 
> I don't think the `.type` check actually guards, maybe thats why.

It doesn't work for me even when I write the exact same code that Space showed above.
![Code_CL55hrVv5M](https://user-images.githubusercontent.com/51517103/124363142-2ff84f80-dc57-11eb-86d7-f51e48e85c7f.png)

Reply:
Seems like it's just me 🙇‍♂️. Updated the second overload to only suggest `cache` for `cacheOptions`.